python/test/unit/geometry/test_bounding_box_tree.py

```diff
@@ -496,3 +498,159 @@ def test_surface_bbtree_collision(dtype):
 
     collisions = compute_collisions_trees(bbtree1, bbtree2)
     assert len(collisions) == 1
+
+
+@pytest.mark.parametrize("dim", [2, 3])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_determine_point_ownership(dim, dtype):
+    """Find point owners (ranks and cells) using bounding box trees + global communication
+    and compare to point ownership data results."""
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank()
+
+    tdim = dim
+    num_cells_side = 4
+    h = dtype(1.0 / num_cells_side)
+    if tdim == 2:
+        mesh = create_unit_square(
+            MPI.COMM_WORLD, num_cells_side, num_cells_side, CellType.quadrilateral, dtype=dtype
```

Comment:
Can it be. parameterised across the mesh? (triangle, quad, tet, hex)?

Reply:
No, as the test is now it needs the elements to coincide with their bounding boxes. I'll commit a version that covers triangles and tet elements.

Reply:
That would be great, thank you.